### ISSUE TITLE
bug: NU wire-port current and the sheet edge classifier wrap at index 0 instead of zero-padding like the solver's own curl

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#689]
## Problem

Two of the non-uniform port/sheet primitives index backwards without a boundary guard, so at index 0 on an axis they read the **far face** of the array instead of zero. The field solver itself does the opposite.

**1. `wire_port_current` (`rfx/nonuniform.py:694-725`, new in PR #684).**

```python
if comp == "ez":
    return ((hy[mi, mj, mk] - hy[mi - 1, mj, mk]) * dual_y
            - (hx[mi, mj, mk] - hx[mi, mj - 1, mk]) * dual_x)
```

With `mi == 0`, `hy[mi - 1]` is `hy[-1]` — plain negative indexing, which wraps. Compare the solver's own curl, which zero-pads explicitly (`rfx/core/yee.py:131-138`):

```python
def _shift_bwd(arr, axis):
    """arr[i-1] with zero at the first position (replaces roll(arr, +1, axis))."""
    padded = jnp.pad(arr, pad_widths)     # (1, 0) on `axis`
    return padded[..., :-1, ...]
```

So a port at index 0 on a transverse axis gets a current that disagrees with what the field update is doing at that same location.

**2. `tangential_edge_masks` (`rfx/boundaries/pec.py:110-119`, from #677).**

```python
mask_ez = cell_mask & (jnp.roll(cell_mask, 1, axis=2) | jnp.roll(cell_mask, -1, axis=2))
```

`jnp.roll` wraps by definition. A one-cell sheet sitting on index 0 (or the last index) of an axis therefore tests adjacency against the opposite face of the domain. If anything is masked there, the sheet-normal component — which must be all-False for a one-cell sheet — picks up spurious True entries, and the PEC/sheet operators then act on an edge they should leave alone.

## Reachability

With CPML pads the physical geometry normally sits well inside, so this is unlikely to fire in an open-domain run. It IS reachable with `boundary='pec'` and `cpml_layers=0` (no pads), which is the shape of several of the closed-box fixtures used to validate the wire-port lane, and for any structure deliberately placed against a domain face. I have not constructed a case that fires it — the defect is read from the source and from the contrast with `_shift_bwd`, not from a failing run.

## Ask

Use the same zero-padded shift the solver uses, in both places, rather than negative indexing or `roll`. A one-line helper already exists (`_shift_bwd` / a matching `_shift_fwd`); reusing it also makes the port's Ampère loop structurally identical to the curl it is supposed to mirror, which is the property #672's docstring is arguing for.

## Falsifier

Place a wire port with a transverse index of 0 in a `boundary='pec'`, `cpml_layers=0` box and compare its reported `I` against the curl the solver applies at the same cell; they must agree. Separately, put a one-cell sheet on index 0 of its normal axis with anything masked on the opposite face and check `mask` for the normal component is still all-False.

Found by an external review pass (Codex) of the merges landed today; item 2 was found while confirming item 1 — same class, different primitive.

R3: memory=1-cell PEC sheet leaves its Ez edge live (a missing/extra sheet edge has turned 50 ohm into 623-5804 ohm here before) | R2-attempts=0 | falsifier=above, not yet run
